### PR TITLE
Rename settings for discovery and repositories

### PR DESCRIPTION
**Pull request by dadoonet**
This PR contains two commits:

* The first one could also be reviewed at #61: it renames settings for repositories 
* The second one renames settings for discovery

Rename settings for repositories
=========================

Storage and Management settings are not related in Azure API.
We were using the following convention for storage specific settings:

```yml
cloud:
    azure:
        storage_account: your_azure_storage_account
        storage_key: your_azure_storage_key
```

A better naming (and Java packaging) would be by moving `storage` settings in their own package
`cloud.azure.storage`:

```yml
cloud:
    azure:
        storage:
            account: your_azure_storage_account
            key: your_azure_storage_key
```

Note that we can still read deprecated settings but it will print a `WARN` with an advice to make sure users don't forget to move.

Rename settings for discovery
=======================

Azure Java SDK follows some conventions about property names and we can use
the same convention. The benefit is that it makes the code cleaner and easier to understand
as components settings relies on package name without the `org.elasticsearch` prefix.

In previous versions, we defined:

```
cloud:
	azure:
		keystore: /path/to/keystore
		password: your_password_for_keystore
		subscription_id: your_azure_subscription_id
		service_name: your_azure_cloud_service_name
		port_name: elasticsearch
		host_type: private_ip
```

We now define:

```
cloud:
	azure:
		management:
			keystore:
				path: /path/to/keystore
				password: your_password_for_keystore
			subscription:
				id: your_azure_subscription_id
			cloud:
				service:
					name: your_azure_cloud_service_name

discovery:
	azure:
		host:
			type: private_ip
		endpoint:
			name: elasticsearch
```

The following are a list of settings (prefixed with `cloud.azure.management`) that can further control the discovery:

* `keystore.path`: /path/to/keystore
* `keystore.password`: your_password for the keystore
* `subscription.id`: your_azure_subscription_id
* `cloud.service.name`: your_azure_cloud_service_name

So basically we deprecate the following properties:

* `cloud.azure.keystore` to `cloud.azure.management.keystore.path`
* `cloud.azure.password` to `cloud.azure.management.keystore.password`
* `cloud.azure.subscription_id` to `cloud.azure.management.subscription.id`
* `cloud.azure.service_name` to `cloud.azure.management.cloud.service.name`
* `cloud.azure.port_name` to `discovery.azure.port.name`
* `cloud.azure.host_type` to `discovery.azure.host.type`

Note that we can still read deprecated settings but it will print a `WARN` with an advice to make sure users don't forget to move.

### CODE REVIEW

**Comment by dadoonet** [comment]
@imotov @tlrx I created here a new PR which contains also PR #61. You can review this one only if you prefer but I found useful to have 2 different commits for those changes. Thanks!

**Comment by dadoonet** [comment]
This PR is included in #64. Closing this one.
